plugins/panel-examples/__init__.py

```diff
@@ -0,0 +1,940 @@
+        ctx.panel.state.plot = [
+            {
+                "values": [19, 26, 55],
+                "labels": ["Residential", "Non-Residential", "Utility"],
+                "type": "pie",
+            }
+        ]
+        ctx.panel.state.plot_layout = {
+            "width": 250,
+            "height": 250,
+            "title": "A Pie Chart",
+        }
+
+    def render(self, ctx: ExecutionContext):
+        # inherit from MultiViewPanel
+        return super().render(ctx)
+
+
+###
+# Inputs
+###
+
+
+class InputMutationsPanel(foo.Panel):
+    @property
+    def config(self):
+        return foo.PanelOperatorConfig(
+            name="example_inputs",
+            label="Python Panel Example: Input Mutations",
+        )
+
+    @staticmethod
+    def on_load(ctx: ExecutionContext):
+        max_images = 10
+        ctx.panel.state.slider_value = 5
+
+        samples = ctx.dataset.limit(max_images)
+        for index, sample in enumerate(samples):
+            image_path = (
+                f"http://localhost:5151/media?filepath={sample.filepath}"
+            )
+            ctx.panel.set_state(f"image{index}", image_path)
+
+    def reset_slider(self, ctx: ExecutionContext):
+        ctx.panel.state.slider_value = 1
+
+    def change_value(self, ctx: ExecutionContext):
+        # grab value from ctx.params
+        ctx.panel.state.slider_value = (
+            ctx.params["value"] or ctx.params["panel_state"]["slider_value"]
+        )
+
+    def render(self, ctx: ExecutionContext):
+        panel = types.Object()
+
+        # load all images in state variable from dataset
+        for index in range(ctx.panel.state.slider_value):
+            image_holder = types.ImageView(width="200px")
+            panel.view(
+                f"image{index}", view=image_holder, caption=f"Image {index}"
+            )
+
+        # Slider
+        schema = {"min": 0, "max": 10, "multipleOf": 1}
+        slider = types.SliderView(
+            data=ctx.panel.state.slider_value, label="Example Slider"
+        )
+
+        panel.int(
+            "slider_value", view=slider, on_change=self.change_value, **schema
+        )
+
+        # Button
+        panel.btn("reset", label="Reset Slider", on_click=self.reset_slider)
+
+        return types.Property(
+            panel,
+            view=types.GridView(
+                align_x="center", align_y="center", orientation="vertical"
+            ),
+        )
+
+
+###
+# Interactive Plot
+###
+
+
+class InteractivePlot(foo.Panel):
+    @property
+    def config(self):
+        return foo.PanelOperatorConfig(
+            name="example_interactive_plot",
+            label="Python Panel Example: Interactive Plot",
+        )
+
+    @staticmethod
+    def on_load(ctx: ExecutionContext):
+
+        # tabulate histogram values
+        label_counts = {}
+        for sample in ctx.dataset.iter_samples():
```

Comment:
Not efficient, should be updated to use `dataset.histogram_values()` or similar

Reply:
How about `ctx.dataset.count_values("ground_truth.detections.label")`?

Reply:
Updated to use `count_values` for now